core/src/main/java/com/datastax/oss/driver/internal/core/tracker/RequestLogger.java

```diff
@@ -155,4 +168,21 @@ protected void logError(
       LOG.error("{} [{}]", builder.toString(), error.toString());
     }
   }
+
```

Comment:
My only question here is should these be moved to a general utility for re-use?

Reply:
Maybe this is too clever but we could generalize this to one method on `DriverConfigProfile` using a function variable for the method to call, i.e.:

```java
public interface DriverConfigProfile {
  // ...
  default <T> Optional<T> get(
      DriverOption option, BiFunction<DriverConfigProfile, DriverOption, T> getter) {
    return isDefined(option) ? Optional.of(getter.apply(this, option)) : Optional.empty();
  }
  // ...
}
```

And to use it:

```java
boolean successEnabled =
    configProfile
        .get(
            DefaultDriverOption.REQUEST_LOGGER_SUCCESS_ENABLED, DriverConfigProfile::getBoolean)
        .orElse(false);
```

Reply:
I don't like how this creates an `Optional` for every call. This is on the hot path because these options are re-read for each request.
Maybe we could have `DriverConfigOption.getBoolean(DriverOption option, boolean default)`, but then it makes sense to have it for every type.

Reply:
To remove optional, tt could be redefined as:

```java
default <T> T get(
      DriverOption option, BiFunction<DriverConfigProfile, DriverOption, T> getter, T defaultValue) {
  return isDefined(option) ? getter.apply(this, option) : defaultvalue;
}
```

But that is more of an effort to reduce adding another method for every possible return type.  Really wish java supported default arguments :rage: